frontend/src/app/sessions/user-sessions-wrapper/active-sessions/active-sessions.component.html

```diff
@@ -106,52 +114,79 @@ <h2>Read-Only session</h2>
               ).info
             "
           >
-            <mat-icon>{{
-              sessionService.beautifyState(
-                session.preparation_state,
-                session.state
-              ).icon
-            }}</mat-icon>
+            <mat-icon
+              >{{
+                sessionService.beautifyState(
+                  session.preparation_state,
+                  session.state
+                ).icon
+              }}
+            </mat-icon>
             {{
               sessionService.beautifyState(
                 session.preparation_state,
                 session.state
               ).text
             }}
           </h3>
-          <p class="mt-1.5">
-            The session was created
-            {{ beautifyService.beatifyDate(session.created_at) }}
-          </p>
-          <div>
-            Tool: {{ session.version.tool.name }} ({{ session.version.name }})
-            <br />
-            Connection Method: {{ session.connection_method?.name }}
-          </div>
+
+          @if (session.idle_state.available) {
+            @let remaining_minutes =
+              session.idle_state.idle_for_minutes! === -1
+                ? session.idle_state.terminate_after_minutes! -
+                  differenceInMinutes(session.created_at, Date.now())
+                : session.idle_state.terminate_after_minutes! -
+                  session.idle_state.idle_for_minutes!;
```

Comment:
That's a lot of logic for the HTML template, I'd prefer to have descriptive function in the ts-file.